clients/src/main/java/org/apache/kafka/common/Metric.java

```diff
@@ -27,8 +27,14 @@
     public MetricName metricName();
 
     /**
-     * The value of the metric
+     * The value of the metric as double if the metric is measurable
+     * @throws IllegalStateException if this metric does not have a measurable double value
      */
     public double value();
```

Comment:
I was wondering if we should deprecate this so that we can eventually replace it with a version that returns `Object`.